src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+            tail = head;
+            size++;
+        } else {
+            node = new Node<>(tail, value, null);
+            node.prev.next = node;
+            tail = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexRangeInAdd(index);
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> node = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            if (node.prev == null) {
+                head = newNode;
+            } else {
+                node.prev.next = newNode;
+            }
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexRange(index);
+        return findNodeByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexRange(index);
+        Node<T> foundNodeByIndex = findNodeByIndex(index);
+        T oldElement = foundNodeByIndex.element;
+        foundNodeByIndex.element = value;
+        return oldElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexRange(index);
+        Node<T> removedNode = findNodeByIndex(index);
+        unlink(removedNode);
+        size--;
+        return removedNode.element;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> removedNodeByValue = findNodeByValue(object);
+        if (removedNodeByValue == null) {
+            return false;
+        } else {
+            unlink(removedNodeByValue);
+            size--;
+            return true;
+        }
+    }
+
+    private Node<T> findNodeByValue(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(currentNode.element, object)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
```

Comment:
The method `findNodeByValue` iterates through the entire list even if the object can be found at the beginning. Consider breaking the loop once the object is found.